src/Workspaces/Core/Portable/FindSymbols/SyntaxTree/SyntaxTreeInfo.cs

```diff
@@ -159,8 +159,19 @@ public static Task<SyntaxTreeDeclarationInfo> GetDeclarationInfoAsync(Document d
                             containsElementAccess = containsElementAccess || syntaxFacts.IsElementAccessExpression(node);
                             containsIndexerMemberCref = containsIndexerMemberCref || syntaxFacts.IsIndexerMemberCRef(node);
 
+                            // We've received a number of error reports where DeclaredSymbolInfo.GetSymbolAsync() will
+                            // crash because the document's syntax root doesn't contain the span of the node returned
+                            // by TryGetDeclaredSymbolInfo().  There are two possibilities for this crash:
+                            //   1) syntaxFacts.TryGetDeclaredSymbolInfo() is returning a bad span, or
+                            //   2) Document.GetSyntaxRootAsync() (called from DeclaredSymbolInfo.GetSymbolAsync) is
+                            //      returning a bad syntax root that doesn't represent the original parsed document.
+                            // By adding the `root.FullSpan.Contains()` check below, if we get similar crash reports in
+                            // the future then we know the problem lies in (2).  If, however, the problem is really in
+                            // TryGetDeclaredSymbolInfo, then this will at least prevent us from returning bad spans
+                            // and will prevent the crash from occuring.
```

Comment:
There's the rub: David and I talked about this at length, and we couldn't see a way to somehow non-fatally track this.  Had we seen this bug 6 months ago we could have thrown an exception/crashed, but since we're trying to stabilize by removing crashes, we couldn't think of anything better.